aws/resource_aws_msk_configuration_test.go

```diff
@@ -2,16 +2,79 @@ package aws
 
 import (
 	"fmt"
+	"log"
 	"regexp"
 	"testing"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/kafka"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/go-multierror"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
+func init() {
+	resource.AddTestSweepers("aws_msk_configuration", &resource.Sweeper{
+		Name: "aws_msk_configuration",
+		F:    testSweepMskConfigurations,
+		Dependencies: []string{
+			"aws_msk_cluster",
+		},
+	})
+}
+
+func testSweepMskConfigurations(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).kafkaconn
+	var sweeperErrs *multierror.Error
+
+	input := &kafka.ListConfigurationsInput{}
+
+	err = conn.ListConfigurationsPages(input, func(page *kafka.ListConfigurationsOutput, isLast bool) bool {
+		if page == nil {
+			return !isLast
+		}
+
+		for _, configuration := range page.Configurations {
+			if configuration == nil {
```

Comment:
not sure the likelihood of catching a configuration in a `deleting` state at this point, but just curious if we should check the state here as well since the docs for `DeleteConfiguration` note the configuration must be in either the `active` or `delete_failed` state?

Reply:
It is certainly in the realm of possibility (albeit very unlikely as these seem to delete very fast currently)! Rather than handling it just in our testing code, it would be something that should be handled in the resource `Delete` function. We won't be able to really test it, but let me see if I can grab the specific error via the AWS CLI.

Reply:
Here's what I got with quickly doing `aws kafka delete-configuration`:

```
An error occurred (BadRequestException) when calling the DeleteConfiguration operation: This operation is only valid for resources that are in one of the following states :[ACTIVE, DELETE_FAILED]
```

I was hoping it would either have a more specific error code for this situation or include `DELETING` in the error message so we could safely skip it, but given the way this error is setup they could add other state values for configurations which we should always return the error instead of skipping it. 🙁  I think we'll want to leave this edge case in there for now unless it does become a real problem for practitioners or our sweepers.